core/src/main/java/io/grpc/internal/ServerCallImpl.java

```diff
@@ -132,9 +132,10 @@ public void sendHeaders(Metadata headers) {
       }
     }
     inboundHeaders = null;
-    if (compressor != Codec.Identity.NONE) {
-      headers.put(MESSAGE_ENCODING_KEY, compressor.getMessageEncoding());
-    }
+
+    // Always put compressor, even if it's identity.
+    headers.put(MESSAGE_ENCODING_KEY, compressor.getMessageEncoding());
```

Comment:
I don't quite follow. I see where you say you are making this change because this is what wrapped languages do, but why does that mean we need any change? It seems both are acceptable, and so we should choose which behavior we prefer.